ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -6,68 +6,93 @@ import {
-  };
-
-  return (
-    <NotificationListItemSnap
-      id={snapNotification.id}
-      snapId={snapNotification.data.origin}
-      isRead={snapNotification.isRead}
-      createdAt={new Date(snapNotification.createdAt)}
-      title={{
-        items: [
-          {
-            text: snapsNameGetter(snapNotification.data.origin) || 'Snap',
-          },
-        ],
-      }}
-      snapMessage={snapNotification.data.message}
-      handleSnapClick={handleSnapClick}
-      handleSnapButton={handleSnapButton}
-    />
-  );
+export const components: NotificationComponent<SnapNotification> = {
+  guardFn: isOfTypeNodeGuard(['snap' as typeof TRIGGER_TYPES.SNAP]),
+  item: ({ notification, onClick }) => {
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const trackEvent = useContext(MetaMetricsContext);
+    const snapsMetadata = useSelector(getSnapsMetadata);
+    const snapsNameGetter = getSnapName(snapsMetadata);
+    const handleSnapButton = () => {
+      dispatch(markNotificationsAsRead([notification.id]));
+      trackEvent({
+        category: MetaMetricsEventCategory.NotificationInteraction,
+        event: MetaMetricsEventName.NotificationClicked,
+        properties: {
+          notification_id: notification.id,
+          notification_type: notification.type,
+          previously_read: notification.isRead,
+        },
+      });
+      history.push(getSnapRoute(notification.data.origin));
+    };
+    return (
+      <NotificationListItemSnap
+        id={notification.id}
+        snapId={notification.data.origin}
+        isRead={notification.isRead}
+        createdAt={new Date(notification.createdAt)}
+        title={{
+          items: [
+            {
+              text: snapsNameGetter(notification.data.origin) || 'Snap',
+            },
+          ],
+        }}
+        snapMessage={notification.data.message}
+        handleSnapClick={onClick}
+        handleSnapButton={handleSnapButton}
+      />
+    );
+  },
+  details: {
+    title: ({ notification }) => (
+      <NotificationDetailTitle
+        title={notification.data.expandedView?.title as string}
+        date={formatIsoDateString(notification.createdAt)}
+      />
+    ),
+    body: {
+      type: 'body_snap_notification',
+      Content: ({ notification }) => (
+        <SnapUIRenderer
+          snapId={notification.data.origin}
+          interfaceId={notification.data.expandedView?.interfaceId as string}
+          useDelineator={false}
+          contentBackgroundColor={BackgroundColor.backgroundDefault}
+        />
+      ),
+    },
+  },
+  footer: {
+    type: 'footer_snap_notification',
+    Link: ({ notification }) =>
+      notification.data.expandedView?.footerLink ? (
+        <NotificationDetailButton
+          notification={notification}
+          text={notification.data.expandedView?.footerLink.text as string}
+          href={notification.data.expandedView?.footerLink.href as string}
```

Comment:
Are these properties guaranteed to be set?

Reply:
Yes, see above.